libbpf-tools/memleak.c

```diff
@@ -0,0 +1,707 @@
+	{ "min-size", 'z', NULL, 0, "Capture only allocations larger than this size" },
+	{ "max-size", 'Z', NULL, 0, "Capture only allocations smaller than this size" },
+	{ "obj", 'O', "OBJ", 0, "Attach to allocator functions in the specified object" },
+	{ "percpu", 'P', NULL, 0, "trace percpu allocations" },
+	{ NULL, 'h', NULL, OPTION_HIDDEN, "Show the full help" },
+	{},
+};
+
+static error_t parse_arg(int key, char *arg, struct argp_state *state)
+{
+	static int pos_args;
+	int pid;
+	int interval;
+	int count;
+	int older;
+	int sample_rate;
+	int min_size;
+	int max_size;
+
+	switch (key) {
+	case 'p':
+		errno = 0;
+		pid = strtol(arg, NULL, 10);
+		if (errno || pid <= 0) {
+			fprintf(stderr, "Invalid pid: %s\n", arg);
+			argp_usage(state);
+		}
+		env.pid = pid;
+		break;
+	case 't':
+		env.trace_all = true;
+		break;
+	case 'a':
+		env.show_allocs = true;
+		break;
+	case 'c':
+		env.command = strdup(arg);
+		break;
+	case 'o':
+		errno = 0;
+		older = strtol(arg, NULL, 10);
+		if (errno || older <= 0) {
+			fprintf(stderr, "Invalid time: %s\n", arg);
+			argp_usage(state);
+		}
+		env.min_age_ns = older * 1e6;
+		break;
+	case 'C':
+		env.combined_only = true;
+		break;
+	case 'W':
+		env.wa_missing_free = true;
+		break;
+	case 's':
+		errno = 0;
+		sample_rate = strtol(arg, NULL, 10);
+		if (errno || sample_rate <= 0) {
+			fprintf(stderr, "Invalid sample rate: %s\n", arg);
+			argp_usage(state);
+		}
+		env.sample_rate = sample_rate;
+		break;
+	case 'z':
+		errno = 0;
+		min_size = strtol(arg, NULL, 10);
+		if (errno || min_size <= 0) {
+			fprintf(stderr, "Invalid min size: %s\n", arg);
+			argp_usage(state);
+		}
+		env.min_size = min_size;
+		break;
+	case 'Z':
+		errno = 0;
+		max_size = strtol(arg, NULL, 10);
+		if (errno || max_size <= 0) {
+			fprintf(stderr, "Invalid max size: %s\n", arg);
+			argp_usage(state);
+		}
+		env.max_size = max_size;
+		break;
+	case 'P':
+		env.percpu = true;
+		break;
+	case 'h':
+		argp_state_help(state, stderr, ARGP_HELP_STD_HELP);
+		break;
+	case ARGP_KEY_ARG:
+		if (pos_args++ > 2) {
+			fprintf(stderr,
+				"Unrecognized positional argument: %s\n", arg);
+			argp_usage(state);
+		}
+		errno = 0;
+		if (pos_args == 1) {
+			interval = strtoll(arg, NULL, 10);
+			if (errno || interval <= 0) {
+				fprintf(stderr, "Invalid report interval: %s\n", arg);
+				argp_usage(state);
+			}
+			env.interval = interval;
+		}
+		if (pos_args == 2) {
+			count = strtoll(arg, NULL, 10);
+			if (errno || count <= 0) {
+				fprintf(stderr, "Invalid number of times to print report: %s\n", arg);
+				argp_usage(state);
+			}
+			env.count = count;
+		}
+		break;
+	default:
+		return ARGP_ERR_UNKNOWN;
+	}
+	return 0;
+}
+
+static const struct argp argp = {
+	.options = opts,
+	.parser = parse_arg,
+	.doc = argp_program_doc,
+};
+
+static int libbpf_print_fn(enum libbpf_print_level level, const char *format, va_list args)
+{
+	return vfprintf(stderr, format, args);
```

Comment:
Check `level` against `verbose` option.

Reply:
Fix in commit [2beb289e](https://github.com/iovisor/bcc/pull/3828/commits/2beb289e98d651bddc2127299c5627b5d6149fce).